packages/backend/src/open_payments/receiver/service.ts

```diff
@@ -298,10 +298,24 @@ async function getIncomingPaymentGrant(
   const existingGrant = await deps.grantService.get(grantOptions)
   if (existingGrant) {
     if (existingGrant.expired) {
-      // TODO
-      // https://github.com/interledger/rafiki/issues/795
-      deps.logger.warn({ grantOptions }, 'Grant access token expired')
-      return undefined
+      if (!existingGrant.managementUrl || !existingGrant.accessToken) {
+        deps.logger.warn({ existingGrant }, 'Grant token cannot be rotated.')
+        return undefined
+      }
+      try {
+        const rotatedToken = await deps.openPaymentsClient.token.rotate({
+          url: existingGrant.managementUrl,
+          accessToken: existingGrant.accessToken
+        })
+        return deps.grantService.update(existingGrant, {
+          accessToken: rotatedToken.access_token.value,
+          managementUrl: rotatedToken.access_token.manage,
```

Comment:
Since this `url` should adhere to the Open Payments spec, could we just store the `managementId`?
https://docs.openpayments.guide/reference/post-token
The url can be reconstructed when rotating with
```
`${grant.authServer.url}/token/${grant.managementId}`
```
(could be a `managementUrl` getter on the `Grant` model)

Reply:
is it possible `${grant.authServer.url}/token/${grant.managementId}` !== `rotatedToken.access_token.manage` in any scenario? If not then `managementId` with a `managementUrl` getter makes sense to me

Reply:
The fact that we've got the rotate and revoke token routes defined in the Open Payments spec makes me think `access_token.manage` should always match that format.
I'm even wondering if `open-payments` client should be validating that in relevant response bodies.

Reply:
I think we should do that 👆

Reply:
I'm guessing that check can be done anytime we return an access_token:
- grant creation
- grant continuation
- token rotation
right?

Reply:
:heavy_check_mark:  right

Reply:
https://github.com/interledger/rafiki/issues/1063